### ISSUE TITLE
Passing CustomFields as array to ViewableData::renderWith doesn't pass data in included templates

### ISSUE THREAD

**Issue by wernerkrauss** [silverstripe/silverstripe-framework#8580]
Affected Version: 4.2

Steps to reproduce:

E.g. with SilverShop, Emails are created with custom data like:

```
        $email->setData(
            [
                'PurchaseCompleteMessage' => $completemessage,
                'Order' => $this->order,
                'BaseURL' => Director::absoluteBaseURL(),
            ]
        );
```

I can call `$Order` in my main email template, but for some reason when I have a template with an include I cannot get the order object in my included template unless I pass it manually:

`<% include OrderEmail Order=$Order %>`

This seems to be related to this 

https://github.com/silverstripe/silverstripe-framework/blob/4d14e9b6b1c7ede1c9a52efd366cafe8d6cc3cdf/src/View/ViewableData.php#L397-L402

If I pass the custom data wrapped in `ArrayData` it works as expected. Is this a bug or a feature?

**Comment by kinglozzer** [comment]
Yep, that is expected behaviour (currently), see Hamish’s explanation for why here: https://github.com/silverstripe/silverstripe-framework/issues/2797#issuecomment-65524823

**Comment by wernerkrauss** [comment]
@kinglozzer but why is `$customFields` working different if it's an `array` or an `ArrayData` / `ViewableData`? It's not just "scope" like `$Top`, it's custom data I pass to the template. 

passing `$customFields` as `ArrayData`: can access it in an included template
passing `$customFields` as `php array`: cannot access it... but why???

Where should we fix this? In ViewableData? In `Email::setData()`? Convert to ArrayData if we pass an array to it?

Or just in our custom code, e.g. silvershop's OrderEmailNotifier?

**Comment by kinglozzer** [comment]
> If I pass the custom data wrapped in `ArrayData` it works as expected.

Sorry @wernerkrauss, I missed that line!

So it looks like `ViewableData` instances get folded into the main `$data` variable, but arrays are just passed as “overrides” which aren’t preserved when jumping down into an include. I think we should make them behave the same when passed to `ViewableData::renderWith()`, could you make a PR against `4` to see if it breaks any unit tests? We’ll probably need to carefully review it anyway but that would be a good starting point 😄
